ouroboros/chainsync/compatibility/types.go

```diff
@@ -75,35 +78,37 @@ func (c *CompatibleResultNextBlock) UnmarshalJSON(data []byte) error {
 	// Assume v6 responses first, then fall back to manual v5 processing.
 	var r chainsync.ResultNextBlockPraos
 	err := json.Unmarshal(data, &r)
-	if err == nil && r.Tip != nil {
+	if err == nil && r.Direction != "" {
```

Comment:
Heh. This change is why I didn't push at 1 AM. I looked and somehow missed `Tip` being the one field in both `Result` types.